test/e2e/scheduling/taint_based_evictions.go

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	schedulerapi "k8s.io/kubernetes/pkg/scheduler/api"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+)
+
+func newUnreachableNoExecuteTaint() *v1.Taint {
+	return &v1.Taint{
+		Key:    schedulerapi.TaintNodeUnreachable,
+		Effect: v1.TaintEffectNoExecute,
+	}
+}
+
+func runTBEController(cs clientset.Interface, podUpdated, podDeleted chan *v1.Pod, stopCh chan struct{}, ns string) {
```

Comment:
Can you please add some comments here for the function? Does TBE refer to `taintBasedEviction` here?

Reply:
Sure, will do.

Yes, TBE = TaintBasedEviction :)